apps/chat/src/store/application/application.epics.ts

```diff
@@ -675,6 +684,70 @@ const resetSelectedWidgetEpic: AppEpic = (action$) =>
     }),
   );
 
+const duplicateApplicationEpic: AppEpic = (action$, state$) =>
+  action$.pipe(
+    filter(ApplicationActions.duplicate.match),
+    switchMap(({ payload }) => {
+      const state = state$.value;
+      const agentsMap = ModelsSelectors.selectModelsMap(state);
+      const myCustomAgents = ModelsSelectors.selectCustomModels(state).filter(
+        (application) => isMyApplication(application),
+      );
+      const agent = agentsMap[payload.reference];
+
+      if (!agent) {
+        return of(
+          UIActions.showErrorToast(
+            'Application does not exist. Try to reload the page.',
+          ),
+        );
+      }
+
+      if (!isApplicationId(agent.id)) {
```

Comment:
Should it be **isMyApplication** function instead of  **isApplicationId**
```suggestion
      if (!isMyApplication(agent)) {
```